packages/web-components/src/components/va-combo-box/va-combo-box.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  forceUpdate,
+  Host,
+  Prop,
+  h,
+  State,
+  Fragment,
+} from '@stencil/core';
+import classnames from 'classnames';
+import { i18next } from '../..';
+import { comboBox } from './va-combo-box-library.js';
+
+/**
+ * @componentName Combo Box
+ * @maturityCategory dont_use
+ * @maturityLevel proposed
+ * @guidanceHref form/combo-box
+ * @translations English
+ * @translations Spanish
+ */
+
+@Component({
+  tag: 'va-combo-box',
+  styleUrl: 'va-combo-box.scss',
+  shadow: true,
+})
+export class VaComboBox {
+  @Element() el: any;
+
+  @State() options: Array<Node>;
+
+  @State() labelNode: HTMLLabelElement;
+
+  /**
+   * Whether or not this is a required field.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * The combo box component will be disabled / read-only.
+   */
+  @Prop() disabled?: boolean = false;
+
+  /**
+   * Text label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * The placeholder string.
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * Name attribute for the select field.
+   */
+  @Prop() name!: string;
+
+  /**
+   * Selected value (will get updated on select).
+   */
+  @Prop({ reflect: true, mutable: true }) value?: string;
+
+  /**
+   * Error message to display. When defined, this indicates an error.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Whether or not `aria-invalid` will be set on the inner select. Useful when
+   * composing the component into something larger, like a date component.
+   */
+  @Prop() invalid?: boolean = false;
+
+  /**
+   * Whether or not to fire the analytics events
+   */
+  @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * Optional hint text.
+   */
+  @Prop() hint?: string;
+
+  /**
+   * An optional message that will be read by screen readers when the select is focused.
+   */
+  @Prop() messageAriaDescribedby?: string;
+
+  /**
+   * The event emitted when the selected value changes
+   */
+  @Event() vaSelect: EventEmitter;
+
+  /**
+   * Whether an error message should be shown - set to false when this component is used inside va-date or va-memorable-date
+   * in which the error for the va-select will be rendered outside of va-select
+   */
+  @Prop() showError?: boolean = true;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * option is selected and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+    if (!this.label) {
+      throw new Error('va-combo-box: label is a required property');
```

Comment:
I'm not sure about this approach for throwing an Error on required props. 🤔  Generally we will just allow the markup for the prop to display or not display.

Reply:
I was not sure how to approach this. But technically `init` function will fail if `<label>` is not present. I decided to keep `label` prop as required and throw a console error if it is not provided so that it is more explicit to why component may not be behaving as expected.

Reply:
@oleksii-morgun  I see what you mean. In the script file, there's also some logic to throw an error for it too. Is that not flagging as I would expect? I just want to make sure we aren't adding duplicate errors.

https://github.com/department-of-veterans-affairs/component-library/blob/1af41e7a68ee8ca524f294356fa6ce4435c31a03/packages/web-components/src/components/va-combo-box/va-combo-box-library.js#L202-L212

Reply:
I just re-tested and realized that at some point I added <label> in markup and now this is irrelevant. I will remove this error from here